modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/history/async/AsyncCmmnHistoryManager.java

```diff
@@ -88,7 +88,20 @@ public void recordUpdateCaseInstanceName(CaseInstanceEntity caseInstanceEntity,
             getAsyncHistorySession().addHistoricData(getJobServiceConfiguration(), CmmnAsyncHistoryConstants.TYPE_UPDATE_CASE_INSTANCE_NAME, data, caseInstanceEntity.getTenantId());
         }
     }
-    
+
+    @Override
+    public void recordUpdateBusinessKey(CaseInstanceEntity caseInstanceEntity, String businessKey) {
+        if (cmmnEngineConfiguration.getHistoryLevel().isAtLeast(HistoryLevel.ACTIVITY)) {
+            ObjectNode data = cmmnEngineConfiguration.getObjectMapper().createObjectNode();
+            putIfNotNull(data, CmmnAsyncHistoryConstants.FIELD_ID, caseInstanceEntity.getId());
+            putIfNotNull(data, CmmnAsyncHistoryConstants.FIELD_NAME, caseInstanceEntity.getBusinessKey());
```

Comment:
This should be `CmmnAsyncHistoryConstants.FIELD_BUSINESS_KEY` instead of `FIELD_NAME`. You can also add a test to `AsyncCmmnHistoryTest` to validate this as well